vendor/src/github.com/docker/engine-api/types/types.go

```diff
@@ -274,6 +274,19 @@ type ExecStartCheck struct {
 	Tty bool
 }
 
+type HealthcheckResult struct {
+	Start, End time.Time
+	ExitCode   int    // 0=healthy, 1=unhealthy, 2=starting
+	Output     string // Output from last check
```

Comment:
If you truncate it you do have to care about the encoding, or you might truncate in the middle of a UTF8 sequence and output invalid bytes later, so stupid as it is I think it might be best to use a string, though I would probably recommend ascii output.